src/groupByKey.ts

```diff
@@ -2,6 +2,24 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
   // write code here;
+
+  const obj: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyObj = item[key] as string;
+
+    if (obj[keyObj] !== undefined && obj[keyObj].length !== 0) {
+      obj[keyObj].push(item);
+    } else {
+      obj[keyObj] = [];
+      obj[keyObj].push(item);
+    }
+  });
+
+  return obj;
```

Comment:
- don't need to create additional variable `obj `, you can use method reduce for in this case
 - `keyObj` is not a correct name, it's better to rename it to `keyValue`